features/release.feature

```diff
@@ -294,3 +294,42 @@ MARKDOWN
       """
       Attaching release asset `hello-1.2.0.tar.gz'...\n
       """
+
+    Scenario: Download a release asset.
+      Given the GitHub API server:
+        """
+        get('/repos/mislav/will_paginate/releases') {
+          json [
+            { url: 'https://api.github.com/repos/mislav/will_paginate/releases/123',
+              upload_url: 'https://api.github.com/uploads/assets{?name,label}',
+              tag_name: 'v1.2.0',
+              name: 'will_paginate 1.2.0',
+              draft: true,
+              prerelease: false,
+              assets: [
+                { url: 'https://api.github.com/repos/mislav/will_paginate/assets/9876',
+                  name: 'hello-1.2.0.tar.gz',
+                },
+              ],
+            },
+          ]
+        }
+        get('/repos/mislav/will_paginate/assets/9876') {
+          status 302
+          headers['Location'] = 'https://github-cloud.s3.amazonaws.com/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz'
+          ""
+        }
+        get('/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz') {
```

Comment:
Strange. That should work, since we're using `:host_name` constraint elsewhere in the test suite. When debugging, I think you should probably look at `request.headers['host']` or `request.env['HTTP_HOST']` since `request.host` is not guaranteed to be accurate.